### PR TITLE
boards: mps2-an385: this platform is supported by qemu

### PR DESCRIPTION
**Pull request by nashif**
In Qemu 2.10 we can run this board configuration in Qemu directly. Make
this possible by setting the board configuration to allow emulation.

Until we have the latest QEMU in the SDK this can be used with a recent
QEMU like this:

$ mkdir build; cd build
$ cmake -DQEMU=/usr/bin/qemu-system-arm -DBOARD=mps2_an385 ..
$ make run

### CODE REVIEW

File: scripts/sanity_chk/sanitycheck-platform-schema.yaml [hunk 1]
```diff
@@ -18,6 +18,9 @@ mapping:
   "type":
     type: str
     enum: [ 'mcu', 'qemu', 'sim', 'unit', 'native']
+  "emulation":
```
Comment (galak) [inline scripts/sanity_chk/sanitycheck-platform-schema.yaml]: Does 'simulation' make more sense instead of 'emulation'?

**Review by galak — DISMISSED** [review]
Some minor comments about xt-sim and using simulation instead of emulation.

I feel like emulation tends to mean HW means of mimicking real hardware (i.e. like using an FPGA or something like palladium from cadence.

File: scripts/sanity_chk/sanitycheck-platform-schema.yaml [hunk 2]
```diff
@@ -18,6 +18,9 @@ mapping:
...
+    type: str
+    enum: [ 'qemu', 'simics' ]
```
Comment (galak) [inline scripts/sanity_chk/sanitycheck-platform-schema.yaml]: Don't we also have xt-sim?

File: scripts/sanity_chk/sanitycheck-platform-schema.yaml (hunk 1 above)
Comment (nashif) [reply scripts/sanity_chk/sanitycheck-platform-schema.yaml]: Doesn't QEMU stand for "Quick Emulator"? 

either way, I do not really care how we call it, the distinction here is needed for real platforms such as the mps2-an385 that can run under an emulator, the binary basically can be flashed to a real device as opposed to some qemu configurations we have which are configured to be run under qemu only.

File: scripts/sanity_chk/sanitycheck-platform-schema.yaml (hunk 2 above)
Comment (nashif) [reply scripts/sanity_chk/sanitycheck-platform-schema.yaml]: yes, probably belong here as well.

**Comment by nashif** [comment]
@galak ping, your change requests have been addressed

**Pull request merged**